scripts/waifulib/xcompile.py

```diff
@@ -434,6 +434,75 @@ def ldflags(self):
 		ldflags = [] # ['-lm', '-lstdc++']
 		return ldflags
 
+class PSVita:
+	ctx          = None # waf context
+	arch         ='armeabi-v7a-hard'
+	vitasdk_dir  = None
+
+	def __init__(self, ctx):
+		self.ctx = ctx
+
+		for i in PSVITA_ENVVARS:
+			self.vitasdk_dir = os.getenv(i)
+			if self.vitasdk_dir != None:
+				break
+		else:
+			ctx.fatal('Set %s environment variable pointing to the VitaSDK directory!' %
+				' or '.join(PSVITA_ENVVARS))
+
+	def gen_toolchain_prefix(self):
+		return 'arm-vita-eabi-'
+
+	def gen_gcc_toolchain_path(self):
+		return os.path.join(self.vitasdk_dir, 'bin', self.gen_toolchain_prefix())
+
+	def cc(self):
+		return self.gen_gcc_toolchain_path() + 'gcc'
+
+	def cxx(self):
+		return self.gen_gcc_toolchain_path() + 'g++'
+
+	def strip(self):
+		return self.gen_gcc_toolchain_path() + 'strip'
+
+	def ar(self):
+		return self.gen_gcc_toolchain_path() + 'ar'
+
+	def pkgconfig(self):
+		return self.gen_gcc_toolchain_path() + 'pkg-config'
+
+	def cflags(self, cxx = False):
+		cflags = []
+		# arch flags
+		cflags += ['-D__vita__', '-mtune=cortex-a9', '-mfpu=neon']
+		# necessary linker flags
+		cflags += ['-Wl,-q', '-Wl,-z,nocopyreloc']
+		# this optimization is broken in vitasdk
+		cflags += ['-fno-optimize-sibling-calls']
+		# disable some ARM bullshit
+		cflags += ['-fsigned-char', '-fno-short-enums', '-Wno-attributes']
+		# base include dir
+		cflags += ['-isystem %s/arm-vita-eabi/include' % self.vitasdk_dir]
+		# SDL include dir
+		cflags += ['-I%s/arm-vita-eabi/include/SDL2' % self.vitasdk_dir]
```

Comment:
SDL2 shouldn't be added globally here and instead be in uselib.

But this is a minor issue, I wouldn't change it right now but refactor it in engine later. Maybe I would do that myself.